module/datamanager.py

```diff
@@ -1000,21 +1053,41 @@ def get_overall_it_state(self, user):
         return hosts_state, services_state
 
     def get_important_elements(self, user, type='all', sorter=worse_first):
-        return self.search_hosts_and_services('bi:>2 ack:false type:%s' % type, user=user, sorter=sorter)
+        # BI is greater than the minimum business impact
+        # todo: @maethor: why searching with ack:false ? an element is important whether it is ack or not...
+        search = "bi:>%d " % (self.min_business_impact) + 'ack:false type:%s' % type
+        return self.search_hosts_and_services(search, user=user, sorter=sorter)
 
-    def get_impacts(self, user, search='is:impact bi:>=0 type:all', sorter=worse_first):
+    def get_impacts(self, user, search='is:impact type:all', sorter=worse_first):
         if "is:impact" not in search:
-            search = "is:impact " + search
+            search = search +  " is:impact"
+        if "bi:" not in search:
+            search = search + " bi:>=%d " % (self.min_business_impact)
         return self.search_hosts_and_services(search, user=user, get_impacts=True, sorter=sorter)
 
-    def get_problems(self, user, search='isnot:UP isnot:OK isnot:PENDING bi:>=0 type:all',
-                     get_acknowledged=False, get_downtimed=False, sorter=worse_first):
+    def get_problems(self, user, search='isnot:UP isnot:OK isnot:PENDING isnot:ACK isnot:DOWNTIME '
+                                        'is:HARD type:all', get_acknowledged=False, get_downtimed=False,
+                     sorter=worse_first):
+        """What are problems:
+        - hosts/services (type:all) that are not OK/UP in a HARD state (isnot:hUP isnot:hOK)
+        - and that are not PENDING
+        - and which business impact is greater than or equal the minimum business impact (defaults 0)
+        """
         if "isnot:UP" not in search:
-            search = "isnot:UP " + search
+            search = search + " isnot:UP"
```

Comment:
Why did you change this? It is way better to filter first on this because it is one of the faster and more discriminent filter.

Reply:
+1 for you! I did a bad copy/paste ... I fix this and I will rebase/squash the commits for you merge this